File: app/src/main/java/marcelo/breguenait/breedingassistant/screens/assistant/AssistantContract.java
package marcelo.breguenait.breedingassistant.screens.assistant;

import java.util.List;

import marcelo.breguenait.breedingassistant.data.internal.InternalPokemon;
import marcelo.breguenait.breedingassistant.logic.CombinationHolder;

/**
 * Created by Marcelo on 12/02/2017.
 */

public interface AssistantContract {

    interface AssistantView {

        /*Goal information update*/
        void updateSelectedPokemonName(String name);

        void updateSelectedPokemonIcon(int iconId);

        void updateSelectedPokemonIVs(int[] IVs);

        void updateSelectedPokemonExtraInfo(String nature, String ability);

        void showCreatePokemon();

        void showEditGoal();

        void showLoading();

        void provideDirectItems(List<CombinationHolder> chances, int flags);

        void provideImprovementItems(List<CombinationHolder> improvements);

    }

    interface StorageView {

        boolean initialized();

        void updateStoredPokemons(List<InternalPokemon> storedPokemons);

        void showSuccessfulStorage();
    }

    interface Presenter {

        void setAssistantView(AssistantView assistantView);

        void setStorageView(StorageView storageView);

        void startAssistant();

        void startStored();

        void result(int requestCode, int resultCode);

        void requestChancesUpdate();

        String getPokemonName(int externalId);

        int getPokemonIconId(int externalId);

        String getNatureName(int natureId);

        String getAbilityName(int pokemonId, int abilitySlot);

        void storeNewPokemon();

        void editGoal();

        InternalPokemon getCurrentGoal();

        void removeStoredPokemons(List<InternalPokemon> stored);

    }
}

File: app/src/main/java/marcelo/breguenait/breedingassistant/screens/assistant/AssistantContract.kt
package marcelo.breguenait.breedingassistant.screens.assistant

import marcelo.breguenait.breedingassistant.logic.CombinationHolder

/**
 * Created by Marcelo on 12/02/2017.
 */

interface AssistantContract {

    interface AssistantView {

        /*Goal information update*/
        fun updateSelectedPokemonName(name: String)

        fun updateSelectedPokemonIcon(iconId: Int)

        fun updateSelectedPokemonIVs(IVs: IntArray)

        fun updateSelectedPokemonExtraInfo(nature: String, ability: String)

        fun showCreatePokemon()

        fun showEditGoal()

        fun showLoading()

        fun provideDirectItems(chances: List<CombinationHolder>, flags: Int)

        fun provideImprovementItems(improvements: List<CombinationHolder>)

        fun runLayoutAnimation()

    }



    interface Presenter: BoxContract.Presenter {

        fun setAssistantView(assistantView: AssistantView)

        fun startAssistant()

        fun requestChancesUpdate()

        fun getPokemonName(externalId: Int): String

        fun getPokemonIconId(externalId: Int): Int

        fun getNatureName(natureId: Int): String

        fun getAbilityName(pokemonId: Int, abilitySlot: Int): String

        fun storeNewPokemon()

        fun editGoal()
    }
}
